src/builddef.c

```diff
@@ -109,30 +109,34 @@ int main(int argc, char **argv) {
 
             char *bootdef_value;
 
-            if ( ( bootdef_value = strtok(NULL, "\n") ) != NULL ) {
-
+            bootdef_value = strtok(NULL, "\n");
+            char empty[] = "";
+            if (bootdef_value == NULL) {
+                bootdef_value = empty;
+            } else {
                 chomp_comments(bootdef_value);
-                singularity_message(VERBOSE2, "Got bootstrap definition key/val '%s' = '%s'\n", bootdef_key, bootdef_value);
-
-                if ( envar_defined(strjoin("SINGULARITY_DEFFILE_", uppercase(bootdef_key))) == 0 ) {
-                    singularity_message(ERROR, "Duplicate bootstrap definition key found: '%s'\n", bootdef_key);
-                    ABORT(255);
-                }
-
-                if ( strcasecmp(bootdef_key, "import") == 0 ) {
-                    // Do this again for an imported deffile
-                    bootstrap_keyval_parse(bootdef_value);
-                }
-
-                if ( strcasecmp(bootdef_key, "bootstrap") == 0 ) {
-                    singularity_registry_set("DRIVER", bootdef_value);
-                }
-
-                // Cool little feature, every key defined in def file is transposed
-                // to environment
-                envar_set(uppercase(bootdef_key), bootdef_value, 1);
-                envar_set(strjoin("SINGULARITY_DEFFILE_", uppercase(bootdef_key)), bootdef_value, 1);
             }
+
+            singularity_message(VERBOSE2, "Got bootstrap definition key/val '%s' = '%s'\n", bootdef_key, bootdef_value);
+
+            if ( envar_defined(strjoin("SINGULARITY_DEFFILE_", uppercase(bootdef_key))) == 0 ) {
+                singularity_message(ERROR, "Duplicate bootstrap definition key found: '%s'\n", bootdef_key);
+                ABORT(255);
+            }
+
+            if ( strcasecmp(bootdef_key, "import") == 0 ) {
+                // Do this again for an imported deffile
+                bootstrap_keyval_parse(bootdef_value);
+            }
+
+            if ( strcasecmp(bootdef_key, "bootstrap") == 0 ) {
+                singularity_registry_set("DRIVER", bootdef_value);
+            }
+
+            // Cool little feature, every key defined in def file is transposed
+            // to environment
+            envar_set(uppercase(bootdef_key), bootdef_value, 1);
+            envar_set(strjoin("SINGULARITY_DEFFILE_", uppercase(bootdef_key)), bootdef_value, 1);
```

Comment:
here too, since these logs get shared and moved around in various places, we need to possibly be conservative and juts report that the variable is defined. Especially if it's environment (more likely to have secrets than deffile)

Reply:
Same as above, I didn't write this code, just unindented the whole block. In this specific case, I'm also not sure where the information leak would be ...